packages/experimental/Avatar/src/useAvatar.ts

```diff
@@ -37,9 +40,53 @@ export const useAvatar = (props: JSAvatarProps): AvatarInfo => {
       activeAppearance,
       image: imageProps,
       badge: badgeProps,
+      initials: _initials,
     },
     state: {
       ...state,
     },
   };
 };
+
+/**
+ * A function which takes a name and returns initials
+ *
+ * @param name
+ * @returns initials. First letter of the word if only one word was provided.
+ * First letters of the first and last words if more words were provided.
+ * Words in braces, titles, special characters, parantheses and dashes should be ignored.
+ */
+export const getInitials = (name: string): string => {
+  if (!name) return '';
+  const nonWordRegExp = /\W+(?<!\S-)/g;
+  const wordsInBracesRegExp = /(\(|\[|\{)\w+(\)|\]|\})/g;
+  let words = name.replace(wordsInBracesRegExp, ' ').replace(nonWordRegExp, ' ').trim().split(' ');
+  words = removeTitlesFromName(words);
```

Comment:
If you split the name before running removeTitlesFromName, then I don't think any multi-word titles would match correctly.

The chance of someone using a multi-word title is low though, so I'm okay with either reworking the logic to recognize multi-word titles correctly /or/ skipping multi-word titles. Could be a good question for PM.

Reply:
Just noting that this is still a potential bug. I'm fine with it being addressed in a follow up PR though